lua/controller.lua

```diff
@@ -502,6 +519,39 @@ local function register_controller()
 	core.register_node("drawers:controller", def)
 end
 
+function controller_tl_on_pull(pos, side, player_name)
+	if not controller_allow_metadata_inventory_take then return false end
+	local meta = core.get_meta(pos)
+	local drawer_net_index = index_drawers(pos)
+	meta:set_string("drawers_table_index", core.serialize(drawer_net_index))
+	local first = nil
+	for index,_ in pairs(drawer_net_index) do
+		if index and index ~= "empty" then
+			first = index
+			break
+		end
+	end
+	if first then
+		local stack = ItemStack(first)
+		stack:set_count(stack:get_stack_max())
+		local taken_stack = drawers.drawer_take_item(drawer_net_index[first]["drawer_pos"], stack)
+		if taken_stack then return taken_stack else return nil end
```

Comment:
Would you please be so nice to split this into multiple lines? Such one-liner-multiple-action lines are not convenient to read.

Reply:
Done

Reply:
I don't see any change.

Reply:
`	if first then
		local stack = ItemStack(first)
		stack:set_count(stack:get_stack_max())
		local drawer_pos = drawer_net_index[first]["drawer_pos"]
		local taken_stack = drawers.drawer_take_item(drawer_pos, stack)
		if taken_stack then return taken_stack else return nil end
	end
`